lucene/core/src/java/org/apache/lucene/store/ByteBuffersDataOutput.java

```diff
@@ -412,7 +413,11 @@ public long ramBytesUsed() {
    * lead to hard-to-debug issues, use with great care.
    */
   public void reset() {
-    blocks.stream().forEach(blockReuse);
+    if (blockReuse != NO_REUSE) {
```

Comment:
This check isn't related to what I saw in the profile, though it looks like it could be an easy win in some cases. The important change is the move from `forEach` to a `for` loop.

Reply:
That .stream() call is actually not necessary there -- you could just pass a closure to blocks.forEach... This should be pretty much equivalent to having a for loop. I don't mind switching to more verbose format, I just wanted be hip and groovy with streams...

Reply:
Good point, I'll change it to `blocks.forEach(blockReuse)`.